test/behaviour/resolution/ResolutionSteps.java

```diff
@@ -22,145 +22,192 @@
-            } else if (query instanceof TypeQLInsert) {
-                tx.query().insert(query.asInsert());
-            } else if (query instanceof TypeQLDefine) {
-                tx.query().define(query.asDefine());
-            } else {
-                throw new ScenarioDefinitionException("Query not handled in ResolutionSteps" + queryStatements);
+    @Then("verify answers are consistent across {int} executions")
+    public static void verify_answers_are_consistent_across_n_executions(int executionCount) {
+        Set<ConceptMap> oldAnswers;
+        oldAnswers = inferenceTx().query().match(typeQLQuery).toSet();
+        for (int i = 0; i < executionCount - 1; i++) {
+            try (TypeDB.Transaction transaction = dataSession().transaction(Arguments.Transaction.Type.READ,
+                                                                            new Options.Transaction().infer(true))) {
+                Set<ConceptMap> answers = transaction.query().match(typeQLQuery).toSet();
+                assertEquals(oldAnswers, answers);
             }
-        });
+        }
     }
 
-    private TypeDB.Session reasonedSession() {
-        TypeDB.Session reasonedSession = sessions.stream()
-                .filter(s -> s.database().name().equals(REASONED_DATABASE))
-                .findAny()
-                .orElse(null);
-        assert reasonedSession != null : "Reasoned session with database name " + REASONED_DATABASE + " does not exist";
-        return reasonedSession;
+    @Then("verify answers are correct")
+    public void verify_answers_are_correct() {
+        correctnessVerifier.verifySoundness(typeQLQuery);
+        correctnessVerifier.verifyCompleteness(typeQLQuery);
     }
 
-    private TypeDB.Session materialisedSession() {
-        TypeDB.Session materialisedSession = sessions.stream()
-                .filter(s -> s.database().name().equals(MATERIALISED_DATABASE))
-                .findAny()
-                .orElse(null);
-        assert materialisedSession != null : "Materialised session with database name " + MATERIALISED_DATABASE + " does not exist";
-        return materialisedSession;
+    @Then("verify answers are sound")
+    public void verify_answers_are_sound() {
+        correctnessVerifier.verifySoundness(typeQLQuery);
     }
 
-    private TypeDB.Transaction reasonedDbTxn() {
-        return getTransaction(reasonedSession());
+    @Then("verify answers are complete")
+    public void verify_answers_are_complete() {
+        correctnessVerifier.verifyCompleteness(typeQLQuery);
     }
 
-    private TypeDB.Transaction materialisedDbTxn() {
-        return getTransaction(materialisedSession());
-    }
+    private static void resetDirectory() throws IOException {
+        if (Files.exists(dataDir)) {
+            System.out.println("Database directory exists!");
+            Files.walk(dataDir).sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
```

Comment:
out of curiosity why do we do it sorted?

Reply:
No idea, this is copied over from the preexisting steps and I didn't want to mess with it!